torch/csrc/jit/codegen/cuda/lower_predicate.cpp

```diff
@@ -81,24 +81,30 @@ class ConditionalFromPredicateModifier {
       TORCH_INTERNAL_ASSERT(conditional != nullptr);
       TORCH_INTERNAL_ASSERT(conditional->isA<kir::Bool>());
 
-      // If predicate is not required
-      if (conditional->isConst() && conditional->value().value()) {
-        TORCH_INTERNAL_ASSERT(ite->elseBody().empty());
+      // If the conditional is a constant, then the IfThenElse is not required
+      if (conditional->isConst()) {
+        const auto exprs_copy = (conditional->value().value())
+            ? ite->thenBody().exprs()
+            : ite->elseBody().exprs();
 
         // Handle internal expressions
-        const auto then_exprs_copy = ite->thenBody().exprs();
-        for (auto expr : then_exprs_copy) {
+        for (auto expr : exprs_copy) {
           handle(expr);
         }
 
         if (for_loops_structure_.empty()) {
-          TORCH_INTERNAL_ASSERT(then_exprs_copy.size() == 1);
-          auto expr = then_exprs_copy.front();
+          // Top-Level of for-loop structure
+          // TODO: Promote scope to IR node
+          // TODO: Replace ite with correct scope node
+          TORCH_INTERNAL_ASSERT(
+              exprs_copy.size() == 1,
```

Comment:
What about when the size is 0? E.g., `conditional` is `false` and there's no else block. We should be able to remove the if-then-else block altogether then, right?

Reply:
Instead of directly removing the ITE, why can't this be handled in the codegen?

Reply:
You mean `CudaKernelGenerator`, right? I think that would be actually better because it should do this anyway and we don't need to duplicate the logic.

Reply:
I removed the elimination logic from lower_predicate and updated the CudaKernelGenerator.